lab_3_generate_by_ngrams/main.py

```diff
@@ -309,6 +479,25 @@ def get_next_token(self, sequence: tuple[int, ...]) -> Optional[list[tuple[int,
 
         In case of corrupt input arguments or methods used return None.
         """
+        if not isinstance(sequence, tuple) or not sequence:
+            return None
+
+        possible_tokens = self._model.generate_next_token(sequence)
+        if possible_tokens is None:
+            return None
+        if not possible_tokens:
+            return []
+
+        possible_tokens_list = list(possible_tokens.items())
+        possible_tokens_list.sort(key=lambda x: x[0], reverse=True)
+        possible_tokens_list.sort(key=lambda x: x[1], reverse=True)
+
+        best_tokens = []
+        for _ in range(self._beam_width):
+            best_tokens.append(possible_tokens_list[0])
+            possible_tokens_list.pop(0)
```

Comment:
I believe that is how list slicing was invented

Reply:
fixed